app/src/main/java/ch/dreipol/rezhycle/fragments/BaseFragment.kt

```diff
@@ -78,4 +86,34 @@ abstract class BaseFragment<B : ViewBinding, V : BaseView> : Fragment(), BaseVie
     internal abstract fun createBinding(): B
 }
 
-interface KeyboardUsingFragment
\ No newline at end of file
+@Suppress("DEPRECATION")
+fun Fragment.updateSystemBarColor(systemBarColor: SystemBarColor?) {
+    if (systemBarColor != null) {
+        val window = requireActivity().window
+        window.statusBarColor = resources.getColor(systemBarColor.color, null)
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R) {
+            if (systemBarColor.light) {
+                window.insetsController?.setSystemBarsAppearance(
+                    WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS,
+                    WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS
+                )
+            } else {
+                window.insetsController?.setSystemBarsAppearance(0, WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS)
+            }
+        } else {
+            if (systemBarColor.light) {
+                window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
+            } else {
+                window.decorView.systemUiVisibility = 0
+            }
+        }
```

Comment:
Could you reduce the complexity by extracting two methods depending on the SDK version?